scripts/packages/bazel.sh

```diff
@@ -61,29 +88,77 @@ function get_realpath() {
     fi
 }
 
-BAZEL_REAL="$(dirname "$(get_realpath "${BASH_SOURCE[0]}")")/bazel-real"
-export BAZEL_REAL
-
-WORKSPACE_DIR="${PWD}"
-while [[ "${WORKSPACE_DIR}" != / ]]; do
-    if [[ -e "${WORKSPACE_DIR}/WORKSPACE" ]]; then
-      break;
+function get_workspace_root() {
+  workspace_dir="${PWD}"
+  while [[ "${workspace_dir}" != / ]]; do
+    if [[ -e "${workspace_dir}/WORKSPACE" ]]; then
+      readonly workspace_dir
+      return
     fi
-    WORKSPACE_DIR="$(dirname "${WORKSPACE_DIR}")"
-done
-readonly WORKSPACE_DIR
+    workspace_dir="$(dirname "${workspace_dir}")"
+  done
+  readonly workspace_dir=""
+}
 
-if [[ -e "${WORKSPACE_DIR}/WORKSPACE" ]]; then
-  readonly WRAPPER="${WORKSPACE_DIR}/tools/bazel"
+get_workspace_root
 
-  if [[ -x "${WRAPPER}" && ! -d "${WRAPPER}" ]]; then
-    exec -a "$0" "${WRAPPER}" "$@"
+readonly wrapper_dir="$(dirname "$(get_realpath "${BASH_SOURCE[0]}")")"
+
+function get_bazel_version() {
+  if [[ -n ${USE_BAZEL_VERSION:-} ]]; then
+    readonly reason="specified in \$USE_BAZEL_VERSION"
+    readonly bazel_version="bazel-${USE_BAZEL_VERSION}"
+  elif [[ -e "${workspace_dir}/.bazelversion" ]]; then
+    readonly reason="specified in ${workspace_dir}/.bazelversion"
+    read -r bazel_version < "${workspace_dir}/.bazelversion"
+    readonly bazel_version="bazel-${bazel_version}"
+  elif [[ -x "${wrapper_dir}/bazel-real" ]]; then
+    readonly reason="automatically selected bazel-real"
+    readonly bazel_version="bazel-real"
+  else
+    readonly reason="automatically selected latest available version"
+    readonly bazel_version="$(basename $(find -H "${wrapper_dir}" -maxdepth 1 -name 'bazel-[0-9]*' -type f | sort -V | tail -n 1) 2>/dev/null)"
   fi
+}
+
+get_bazel_version
+
+if [[ -z $bazel_version ]]; then
+  (color "31" "ERROR: No installed Bazel version found, cannot continue."
+  echo ""
+  echo "Workspace root is: ${workspace_dir:-<empty>}"
+  echo "Selected Bazel version is: ${bazel_version:-<empty>} (${reason:-<empty>})"
+  echo ""
+  echo "You can specify which Bazel version to use using these methods:"
+  echo "1. Set \$USE_BAZEL_VERSION to a version number (e.g. export USE_BAZEL_VERSION=0.29.1)."
+  echo "2. Add a .bazelversion file that contains a version number next to your WORKSPACE file."
+  echo "3. Otherwise, the latest Bazel version installed as ${wrapper_dir}/bazel-* will be used."
+  echo ""
+  echo "For case 1 and 2, a binary called \"bazel-\$version\" (e.g. \"bazel-0.29.1\") must be present in ${wrapper_dir}."
+  echo ""
+  echo "You might be able to install specific Bazel versions via apt-get:"
+  echo "  $ apt-get update && apt-get install bazel-0.29.1") >&2
```

Comment:
This should be: `apt-get install bazel=0.29.1` when the multiple version support is available.

Reply:
Unfortunately apt-get doesn't allow users to install multiple versions of a package like this (this would instead replace the current Bazel version with 0.29.1 on the system).

I think we have to create separate side-by-side installable Bazel packages for this.

Reply:
https://github.com/bazelbuild/bazel/pull/10278 will make both possible!